control/control.go

```diff
@@ -637,3 +608,15 @@ func cacheOptKey(opt controlapi.CacheOptionsEntry) (string, error) {
 	}
 	return fmt.Sprint(opt.Type, ":", hash), nil
 }
+
+type roContentStore struct {
+	content.Store
+}
+
+func (cs *roContentStore) Writer(ctx context.Context, opts ...content.WriterOpt) (content.Writer, error) {
+	return nil, errors.Errorf("read-only content store")
+}
+
+func (cs *roContentStore) Delete(ctx context.Context, dgst digest.Digest) error {
+	return errors.Errorf("read-only content store")
+}
```

Comment:
Maybe we should also ban `Manager.Update`? Could also be good to do `IngestManager.Abort`, since that could interfere with upload operations.